### ISSUE TITLE
[Holding on #53718] [$250] Android - Chat - Message with markdowns is pasted without them on Android.

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53832]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.73-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**Email or phone of affected tester (no customers):** ibellicotest+86@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Prerequisite: Account opened in two different devices or environments.
1. Open the Expensify app.
2. Switch to another account opened in different environment.
3. Open any chat.
4. Write a message with markdowns and send it,
5. Long tap over the message and tap on "Copy to Clipboard"
6. Switch to Android app again.
7. Open any chat,
8. Long tap on compose box and paste the message.
9. Verify that markdowns remain on the message when pasted.

## Expected Result:
When a message with markdowns is copied from a different environment, the markdowns should remain when the message is pasted on Android App.

## Actual Result:
Message is pasted without the markdowns when copied from an account opened on a different environment and pasted on Android App.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/50dd36e9-a7ab-4b9b-a8ed-f17b54fa16ca

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866965157173153205</li>
        <li>Upwork Job ID: 1866965157173153205</li>
        <li>Last Price Increase: 2024-12-25</li>
    </ul>
</details>

**Comment by prakashbask** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - Message with markdowns is pasted without them on Android.

### What is the root cause of that problem?
In setClipboardMessage, there is a check to convert only anchor tags to markdown others are ignored and hence the markdown tags are lost as only plain text is copied to Clipboard

https://github.com/Expensify/App/blob/f481d64b7e894b6a5a98dcaec0e48876a42e60f9/src/pages/home/report/ContextMenu/ContextMenuActions.tsx#L52

### What changes do you think we should make in order to solve the problem?
The check for anchor tag can be removed in 
https://github.com/Expensify/App/blob/f481d64b7e894b6a5a98dcaec0e48876a42e60f9/src/pages/home/report/ContextMenu/ContextMenuActions.tsx#L52 
The updated code will be
```
    //  const anchorRegex = CONST.REGEX_LINK_IN_ANCHOR;
    //  const isAnchorTag = anchorRegex.test(content);
    //  const plainText = isAnchorTag ? Parser.htmlToMarkdown(content) : Parser.htmlToText(content);
        const plainText = Parser.htmlToMarkdown(content)
        Clipboard.setHtml(content, plainText);
```
The paste logic in the hook should be updated to handle markdown
https://github.com/Expensify/App/blob/c8e8517e872c6a9dd739d2745ddd302058ac64de/src/hooks/useHtmlPaste/index.ts#L81

If the text contains markdown, then convert it to plain text. The markdownToText function can be designed similarly to htmlToText in

https://github.com/Expensify/expensify-common/blob/706481004516cddf08b7652afdf5db951443fbea/lib/ExpensiMark.ts#L1203

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
We should write test to handle html, markdown and plain text with scenarios related to different environments (web, native, mWeb, desktop)

### What alternative solutions did you explore? (Optional)

**Comment by prakashbask** [comment]
> @prakashbask As I can see this issue relevant not only for Android but for whole mobile platforms (native+web) Could you clarify why it's not reproduce if we copy text from mobile-web and paste to web, we will have text with Markdown But it's reproduce if we copy text from web and paste to mobile-web, we wouldn't have text with Markdown

@ZhenjaHorbach I think you are the facing the issue as described in [53718](https://github.com/Expensify/App/issues/53718). Presently, only Anchor tag or link markdown copy paste is supported from web/mobile-web. However, that feature does not work correctly due to this issue.

**Comment by Pujan92** [comment]
> we are going to fix this in one issue?

Yes @ZhenjaHorbach, both issue seems to require the same solution. So I think we can hold this issue until the other issue gets resolved.

**Comment by garrettmknight** [comment]
Holding this one to see if https://github.com/Expensify/App/issues/53718 fixes it.

**Comment by garrettmknight** [comment]
Still on hold for https://github.com/Expensify/App/issues/53718

**Comment by garrettmknight** [comment]
Still on hold for https://github.com/Expensify/App/issues/53718

**Comment by garrettmknight** [comment]
Still on hold for https://github.com/Expensify/App/issues/53718

**Comment by garrettmknight** [comment]
Still on hold for https://github.com/Expensify/App/issues/53718

**Comment by garrettmknight** [comment]
Still on hold for https://github.com/Expensify/App/issues/53718

**Comment by garrettmknight** [comment]
Still on hold for https://github.com/Expensify/App/issues/53718
